src/service.ts

```diff
@@ -269,20 +269,25 @@ export class Service {
     output.setData(result);
   }
 
-  static createGist(json: object): Promise<string> {
-    return new Promise((resolve, reject) => {
-      const xhr = new XMLHttpRequest();
-      xhr.addEventListener("load", function() {
-        const jsonURI = JSON.parse(this.response).html_url;
-        resolve(jsonURI);
-      });
-      xhr.addEventListener("error", function() {
-        reject();
-      });
-      xhr.open("POST", "https://api.github.com/gists", true);
-      xhr.setRequestHeader("Content-type", "application/json; charset=utf-8");
-      xhr.send(JSON.stringify(json));
-    });
+  static createGist(json: object): any {
+    const url = "https://api.github.com/gists";
+    fetch(url,
+      {headers: new Headers({ "Content-type": "application/json; charset=utf-8" })})
+      .then(response => {
```

Comment:
make method async: `static async createGist(json: object): Promise<any> {` and use await instead of `.then()`